### PR TITLE
Some additions 2

### PR DESCRIPTION
**Pull request by pierre-kci**

### CODE REVIEW

**Comment by ahmednfwela** [comment]
can you please change the screenshot in the doc folder and update readme

File: lib/src/gantt_default_week_header.dart [hunk 1]
```diff
@@ -7,6 +7,17 @@ class GanttChartDefaultWeekHeader extends StatelessWidget {
...
 
+  int get weekNumberWithinThisYear {
+    final numberWeek = (DateTime.utc(
+                    weekDate.year, weekDate.month, weekDate.day)
+                .difference(DateTime.utc(weekDate.year, 1, 1, 0, 0, 0, 0, 0))
+                .inDays /
+            7)
+        .ceil();
+
+    return numberWeek == 0 ? 1 : numberWeek;
```
Comment (ahmednfwela) [inline lib/src/gantt_default_week_header.dart:18]: should be `return numberWeek + 1;`

File: example/lib/main.dart [hunk 2]
```diff
@@ -113,9 +107,9 @@ class _MyHomePageState extends State<MyHomePage> {
               weekEnds: const {WeekDay.friday, WeekDay.saturday},
               isExtraHoliday: (context, day) {
                 //define custom holiday logic for each day
-                return DateUtils.isSameDay(DateTime(2022, 7, 1), day);
+                return DateUtils.isSameDay(DateTime(2023, 1, 1), day);
               },
-              startOfTheWeek: WeekDay.sunday,
+              startOfTheWeek: WeekDay.monday,
```
Comment (ahmednfwela) [inline example/lib/main.dart:112]: should be sunday

**Review by ahmednfwela — CHANGES_REQUESTED** [review]

File: lib/src/gantt_default_week_header.dart (hunk 1 above)
Comment (pierre-kci) [reply lib/src/gantt_default_week_header.dart]: https://www.epochconverter.com/weeks/2023

Very Strange but based on link above it should not actually be incremented at all :)

File: lib/src/gantt_default_week_header.dart [hunk 3]
```diff
@@ -7,6 +7,17 @@ class GanttChartDefaultWeekHeader extends StatelessWidget {
...
+  int get weekNumberWithinThisYear {
+    final numberWeek = (DateTime.utc(
+                    weekDate.year, weekDate.month, weekDate.day)
+                .difference(DateTime.utc(weekDate.year, 1, 1, 0, 0, 0, 0, 0))
+                .inDays /
+            7)
+        .ceil();
+
+    return numberWeek;
+  }
```
Comment (ahmednfwela) [inline lib/src/gantt_default_week_header.dart:19]: I found this library using the correct iso calculations
https://github.com/KadaDev/week_of_year/blob/master/lib/date_week_extensions.dart

File: test/week_number_within_this_year_test.dart [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+
+    final widgetMidOfYear = GanttChartDefaultWeekHeader(
+      weekDate: DateTime(2023, 6, 2, 1, 1, 1), // 2nd of June 2023
+    );
+
+    final widgetEndOfYear = GanttChartDefaultWeekHeader(
+      weekDate: DateTime(2022, 12, 31, 23, 59, 59),
+    );
+
+    expect(0, widgetStartOfYear.weekNumberWithinThisYear);
```
Comment (ahmednfwela) [inline test/week_number_within_this_year_test.dart:23]: apparently 0 means last week of previous year, and it should be changed to the ISO standard

**Review by ahmednfwela — CHANGES_REQUESTED** [review]

File: lib/src/gantt_default_week_header.dart (hunk 3 above)
Comment (pierre-kci) [reply lib/src/gantt_default_week_header.dart:19]: Impressive, thanks for sharing this

File: test/week_number_within_this_year_test.dart (hunk 4 above)
Comment (pierre-kci) [reply test/week_number_within_this_year_test.dart:23]: update in next commit

**Comment by ahmednfwela** [comment]
LGTM can you pull latest version from master to resolve conflicts?

**Comment by ahmednfwela** [comment]
also after this PR I will try to update the algorithm to adapt to custom start of week

**Comment by pierre-kci** [comment]
```
git fetch origin
git merge origin/master
```

Then I get "Already up to date."
Could you please guide me on how to reconcile them ?

**Comment by ahmednfwela** [comment]
1) make sure you are on your branch https://github.com/pierre-kci/flutter_gantt_chart/tree/somae-additions-2
2) look for this remote `git@github.com:Bdaya-Dev/flutter_gantt_chart.git` in `git remote -v`, if it doesn't exist create it, lets call it `package_remote`
3) `git fetch package_remote`
4) `git pull package_remote`

**Comment by pierre-kci** [comment]
I've just merged and push, could you please review @ahmednfwela 
Thank you for providing the command to fetch 
I understand I should fetch from upstream and not from origin

**Comment by pierre-kci** [comment]
Hi @ahmednfwela could you please review the PR I fetched last changes and fix conflicts, should be good now

**Comment by ahmednfwela** [comment]
the PR only shows empty commit, and is still in conflict, would you mind if I created the PR myself?

**Comment by pierre-kci** [comment]
Sorry @ahmednfwela I forgot to checkout, it is ok now no more conflict, thanks for bearing with my clumsy gitiness, work in progress

**Review by ahmednfwela — APPROVED** [review]
LGTM

**Pull request merged**